src/sema/ast.rs

```diff
@@ -133,6 +133,7 @@ pub struct Function {
     pub symtable: Symtable,
     // What events are emitted by the body of this function
     pub emits_events: Vec<usize>,
+    pub target: Target,
```

Comment:
Presumably you're copying in the compilation target into the Function struct since you don't have it available at the various selector() call sites.

It might be better to calculate the selector in Function::new() and then keep the selector as a member. This prevents the selector being recomputed every time (for metadata generation and code generation for example), and also there is no need to keep a copy of the target around.

Reply:
Yes，i agree with you.